ci-operator/step-registry/aws/provision/route53/private-hosted-zone/aws-provision-route53-private-hosted-zone-commands.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+export AWS_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/.awscred"
+REGION="${LEASED_RESOURCE}"
+CLUSTER_NAME="${NAMESPACE}-${JOB_NAME_HASH}"
+
+ROUTE53_HOSTED_ZONE_NAME="${CLUSTER_NAME}.${BASE_DOMAIN}"
+VPC_ID=$(cat "${SHARED_DIR}/vpc_id")
+CALLER_REFERENCE_STR=$ROUTE53_HOSTED_ZONE_NAME
+OUTPUT_ROUTE53_JSON=${SHARED_DIR}/route53_hosted_private_zone.json
+
+echo -e "creating route53 hosted zone: ${ROUTE53_HOSTED_ZONE_NAME}"
+aws --region "$REGION" route53 create-hosted-zone --name "${ROUTE53_HOSTED_ZONE_NAME}" --vpc VPCRegion="${REGION}",VPCId="${VPC_ID}" --caller-reference "${CALLER_REFERENCE_STR}" > "${OUTPUT_ROUTE53_JSON}.creating"
+
+HOSTED_ZONE_ID="$(jq -r '.HostedZone.Id' "${OUTPUT_ROUTE53_JSON}.creating" | awk -F / '{printf $3}')"
+# save hosted zone information to ${SHARED_DIR} for deprovision step
+echo "${HOSTED_ZONE_ID}" > "${SHARED_DIR}/hosted_zone_id"
+CHANGE_ID=$(jq -r '.ChangeInfo.Id' "${OUTPUT_ROUTE53_JSON}.creating" | awk -F / '{printf $3}')
+
+aws --region "${REGION}" route53 wait resource-record-sets-changed --id "${CHANGE_ID}" &
+wait "$!"
+echo "Hosted zone ${HOSTED_ZONE_ID} successfully created."
+
+aws --region "${REGION}" route53 get-hosted-zone --id "${HOSTED_ZONE_ID}" > "${OUTPUT_ROUTE53_JSON}"
+cp "${OUTPUT_ROUTE53_JSON}" "${ARTIFACT_DIR}/"
```

Comment:
nit: no need for this separate copy, you can just `> "${SHARED_DIR}/route53_hosted_private_zone.json"` in the previous line.

Reply:
Done